src/Symfony/Component/Security/Http/Authenticator/JsonLoginAuthenticator.php

```diff
@@ -147,7 +148,7 @@ private function getCredentials(\stdClass $data): array
             $credentials['username'] = $this->propertyAccessor->getValue($data, $this->options['username_path']);
 
             if (!\is_string($credentials['username']) || '' === $credentials['username']) {
-                throw new BadRequestHttpException(\sprintf('The key "%s" must be a non-empty string.', $this->options['username_path']));
+                throw new BadCredentialsException('The key "username" must be a non-empty string.');
```

Comment:
the default for option username_path is "username", so I'm missing what this PR provides. The resulting message will be exactly the same to me. And when one changes the value of the option, then the message might be considered less accurate actually.